torch/csrc/autograd/profiler.cpp

```diff
@@ -16,29 +14,27 @@ void RecordFunction::pushFunctionRange(Function* fn) {
   pushRange(fn->name());
 }
 
-void enableProfiler(bool use_nvprof, bool use_cuda) {
+void enableProfiler(ProfilerState new_state) {
+  TORCH_ASSERT(new_state != ProfilerState::Disabled);
 #ifndef WITH_CUDA
-  if (use_nvprof)
-    throw std::runtime_error("Can't use CUDA profiler - PyTorch was compiled without CUDA");
+  if (new_state == ProfilerState::NVTX)
+    throw std::runtime_error("Can't use NVTX profiler - PyTorch was compiled without CUDA");
 #endif
-  if (profiling) {
-    if (use_nvprof != using_nvprof)
-      throw std::runtime_error("can't change use_nvprof flag while profiler is running");
-    return;
+  if (state != ProfilerState::Disabled && new_state != state) {
```

Comment:
I don't think we want to allow using `enableProfiler` to disable it. Can you please make it an error?